File: sprint3/By lessons/lesson1code1/lesson1code1.java
import java.io.File;
import java.util.LinkedList;
import java.util.Queue;

public class Main {
    public static String find_file(String folder, String file_to_find) {
        Queue<String> queue = new LinkedList<>();
        queue.add(folder);
        while (!queue.isEmpty()) {
            String item = queue.remove();
            File file = new File(item);
            if (file.isDirectory()) {
                File[] files = file.listFiles();
                for (File f : files) {
                    queue.add(f.getAbsolutePath());
                }
            } else {
                if (file.getName().equals(file_to_find)) {
                    return file.getAbsolutePath();
                }
            }
        }
        return null;
    }

    public static void main(String[] args) {
        String folder = "/path/to/folder";
        String file_to_find = "file.txt";
        String result = find_file(folder, file_to_find);
        if (result == null) {
            System.out.println("File not found");
        } else {
            System.out.printf("File found at %s\n", result);
        }
    }
}

File: sprint3/By lessons/lesson1code1/lesson1code1.cpp
#include <iostream>
#include <queue>
#include <filesystem>

namespace fs = std::filesystem;

std::string find_file(const std::string& folder, const std::string& file_to_find) {
    std::queue<std::string> q;
    q.push(folder);
    while (!q.empty()) {
        std::string path = q.front();
        q.pop();
        if (fs::is_directory(path)) {
            for (const auto& item : fs::directory_iterator(path)) {
                q.push(item.path().string());
            }
        } else {
            if (fs::path(path).filename() == file_to_find) {
                return path;
            }
        }
    }
    return "";
}

int main() {
    std::string folder = "/path/to/folder";
    std::string file_to_find = "file.txt";
    std::string result = find_file(folder, file_to_find);
    if (result.empty()) {
        std::cout << "File not found\n";
    } else {
        std::cout << "File found at " << result << "\n";
    }
    return 0;
}
